.github/workflows/api-compat-verification.yml

```diff
@@ -0,0 +1,33 @@
+name: API compatibility verification
+
+on:
+  pull_request:
+    types: [ opened, synchronize, reopened, labeled, unlabeled ]
```

Comment:
Is there any reason to specify these types? Our other workflows look like they apply to all types

Reply:
I copied the job definition from [**changelog-verification.yml**](https://github.com/awslabs/smithy-kotlin/blob/b2350449f70a78f3c4c647f251b49d1cfc0b57e4/.github/workflows/changelog-verification.yml) which also explicitly specifies PR events. I think, in general, we could be more conservative in the subtypes of events which trigger action runs and reduce some of the churn/cost associated with them.

Do you have a reason to prefer running on all event subtypes?

Reply:
No actually, I think the current `types` cover all the cases where this should run, so it's good to leave it as is. I just never saw these types specified (I did not look at the `changelog-verification.yml` job) and was curious

Reply:
In practice for us these are likely about the same but it's fine as is, here is the [full list](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request) for reference if you leave it unspecified and just have `pull_request:`